libtransmission/CMakeLists.txt

```diff
@@ -216,11 +216,11 @@ target_compile_definitions(${TR_NAME}
         PACKAGE_DATA_DIR="${CMAKE_INSTALL_FULL_DATAROOTDIR}"
         $<$<BOOL:${WITH_INOTIFY}>:WITH_INOTIFY>
         $<$<BOOL:${WITH_KQUEUE}>:WITH_KQUEUE>
-        $<$<BOOL:${ENABLE_UTP}>:WITH_UTP>
         $<$<VERSION_LESS:${MINIUPNPC_VERSION},1.7>:MINIUPNPC_API_VERSION=${MINIUPNPC_API_VERSION}> # API version macro was only added in 1.7
         $<$<BOOL:${USE_SYSTEM_B64}>:USE_SYSTEM_B64>
         $<$<BOOL:${HAVE_SO_REUSEPORT}>:HAVE_SO_REUSEPORT=1>
     PUBLIC
+        $<$<BOOL:${ENABLE_UTP}>:WITH_UTP>
```

Comment:
To quote myself,

> Since `WITH_UTP` is also defined as `PRIVATE` for libtransmission (which is correct and shouldn't be changed IMO, because it's only being used in private headers)

Reply:
I just don't understand this logic. You say that "it's only being used in private headers" but that's not true in practice. The shape (for memory allocation purposes) for the private classes matter, and those are ultimately exposed. It is de facto exposed, indirectly, due to the very problem you first raised. Am I just not grok'ing this? Why is it so bad to make the flag PUBLIC? That seems much clearer since it affects the "shape" of the exposed classes;  even if those classes are intended to be internal, they are de facto exposed.

From my pov, libtransmission is intended to be used as a *co-compiled* library, and thus a fair amount of intimacy with the classes is expected. It is not meant to be stable ABI.  That is different from a library delivered as a dynamic loadable with a stable ABI.  I don't quite understand how the current intention. 

I personally find the whole concept of ABI for C++ code hard to stomach without some form of introspection support, via COM or other background object model. But to each his own.

Given I am at the limits of the meager expertise, can you just give me explicit guidance on what to do with this flag. If moving it from PRIVATE -> PUBLIC is bad mojo, what is better?  Other callers (Torrential, eg) will have the same problem with ifdef'd members problem that the unit-tests currently have unless you make the flag PUBLIC or INTERFACE.